### DISCUSSION TITLE
How to replicate Ostris AI toolkit sampler in sdnext?

### DISCUSSION THREAD

**Discussion by SAC020** [vladmandic/sdnext#4535] [Q&A]
Not logging this as a bug because I'm not sure if it's a bug or a configuration issue.

While training loras with AI Toolkit (Qwen 2512 in this case, quantized with an "accuracy recovery adapter") I find myself unable to replicate the training sample outputs in sdnext. 

The lora seems to work perfect in the AI Toolkit samples, but it's performing very poorly in sdnext.

To test it more to the root cause, I did a comparison of outputs at zero training steps (pre-training baseline, core model, no lora yet):

**AI Toolkit**

Prompt: **sacbf, candid documentary photo, a woman**   (sacbf will be the future token, but it's pre-training)

<img width="1024" height="1024" alt="image" src="https://github.com/user-attachments/assets/f0766ad0-c576-4526-a03b-da7b26598a60" />

        seed: 4245461
        guidance_scale: 4
        sample_steps: 25
        noise_scheduler: "flowmatch"
        name_or_path: "Qwen/Qwen-Image-2512"
        quantize: true
        quantize_te: true
        arch: "qwen_image:2512"
        low_vram: true
        model_kwargs: {}
        layer_offloading: false
        qtype: "uint4|ostris/accuracy_recovery_adapters/qwen_image_2512_torchao_uint4.safetensors"
        qtype_te: "qfloat8"

The same prompt/seed/guidance scale/steps in **sdnext**, on-the-fly sdnq quantization int6. Zero resemblance, besides the greenish background.

<img width="1024" height="1024" alt="image" src="https://github.com/user-attachments/assets/82ca3f83-98f7-40af-a9b0-a50bd790d529" />

```
13:39:53-983284 DEBUG    Sampler: "Euler FlowMatch" class=FlowMatchEulerDiscreteScheduler config={'num_train_timesteps':
                         1000, 'shift': 3, 'use_dynamic_shifting': True, 'base_shift': 0.5, 'max_shift': 1.15,
                         'base_image_seq_len': 256, 'max_image_seq_len': 4096, 'time_shift_type': 'exponential'}
13:39:53-985268 INFO     Processing modifiers: apply
13:39:53-990228 INFO     Base: pipeline=QwenImagePipeline task=TEXT_2_IMAGE batch=1/1x1 set={'prompt': 1,
                         'negative_prompt': 1, 'guidance_scale': 4, 'generator': 'cuda:[4245461]',
                         'num_inference_steps': 25, 'true_cfg_scale': 0, 'output_type': 'np', 'width': 1024, 'height':
                         1024, 'parser': 'fixed'}
13:39:54-000644 DEBUG    Encode: prompt="['sacbf, candid documentary photo, a woman\n']" hijack=True
Progress  4.71s/it ████████████▏                       36% 9/25 00:42 01:15 Base13:40:39-419088 TRACE    Server: alive=True requests=844 memory=46.52/63.92 status='idle' task='' timestamp=None
                         current='task(1k9s528f94rjgez)' id='' job=0 jobs=0 total=52 step=9 steps=0 queued=0 uptime=1226
                         elapsed=43.62 eta=None progress=0
Progress  4.69s/it █████████████████████████████████ 100% 25/25 01:57 00:00 Base
13:41:54-160646 DEBUG    Decode: vae=AutoencoderKLQwenImage slicing=False tiling=False latents=[1, 16, 1, 128,
                         128]:cuda:0 dtype=torch.bfloat16 time=1.231
```

Any idea how to approach this?

**Comment by SAC020** [comment]
To simplify more: Z-image turbo, no quantization at all 

**AI Toolkit**


```
        noise_scheduler: "flowmatch"
          - prompt: "full body view photo of a hiker woman on a mountain trail wearing hiking equipment and boots"
        neg: ""
        seed: 4245456
        walk_seed: false
        guidance_scale: 1
        sample_steps: 8
```

<img width="1024" height="1024" alt="image" src="https://github.com/user-attachments/assets/e42bcc77-05b4-4d26-a986-d6e252e4702e" />


.

**SD-Next** - Z-image turbo, no quantization



It's closer, but far from identical. Default sampler behaves identical with EulerFlowmatch in SD-Next



```
Prompt: full body view photo of a hiker woman on a mountain trail wearing hiking equipment and boots

Parameters: Steps: 8| Size: 1024x1024| Sampler: Euler FlowMatch| Seed: 4245456| CFG scale: 1.0| App: SD.Next| Version: 47dcab3| Pipeline: ZImagePipeline| Operations: txt2img| Model: Z-Image-Turbo| Sampler low order: False| Sampler dynamic shift: True

Time: 11.55s | total 23.21 pipeline 7.72 decode 3.37 onload 3.26 preview 2.99 offload 2.78 te 1.63 callback 0.88 post 0.41 | GPU 14270 MB 44% | RAM 43.29 GB 68%
```

<img width="1024" height="1024" alt="image" src="https://github.com/user-attachments/assets/3ff64721-1a9c-43e5-8a0a-3af3cb5d0eac" />
